diff --git a/gcc/collect2.c b/gcc/collect2.c
index 33206f5c..4068956b 100644
--- a/gcc/collect2.c
+++ b/gcc/collect2.c
@@ -1534,7 +1534,7 @@ collect_execute (const char *prog, char **argv, const char *redir)
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, 0755);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
